pkg/services/dashboards/database/database.go

```diff
@@ -962,17 +965,236 @@ func (d *dashboardStore) FindDashboards(ctx context.Context, query *dashboards.F
 
 	sql, params := sb.ToSQL(limit, page)
 
+	// Only use modified search for non-empty search queries, otherwise it's just listing and new query can't help there yet.
+	resc := make(chan []dashboards.DashboardSearchProjection, 1)
+	if d.features.IsEnabled(ctx, featuremgmt.FlagSearchAlt) && d.features.IsEnabled(ctx, featuremgmt.FlagSplitScopes) &&
+		query.Title != "" && len(query.FolderUIDs) == 0 && len(query.FolderIds) == 0 { //nolint:staticcheck
+		derivedCtx := context.WithoutCancel(ctx)
+		go func() {
+			start := time.Now()
+			results, err := d.findDashboards(derivedCtx, query)
+			if err != nil {
+				d.log.Info("Alternative search query failed", "error", err)
+				return
+			}
+			d.log.Info("Alternative search query", "time", time.Since(start), "results", len(results))
+
+			expected := <-resc
+
+			expectedUIDs, resultUIDs, n, m := map[string]struct{}{}, map[string]struct{}{}, 0, 0
+			for _, hit := range expected {
+				expectedUIDs[hit.UID] = struct{}{}
+			}
+			for _, hit := range results {
+				resultUIDs[hit.UID] = struct{}{}
+				if _, ok := expectedUIDs[hit.UID]; !ok {
+					d.log.Info("Alternative search query mismatch", "unexpected", hit.Title, "uid", hit.UID)
+					n++
+				}
+			}
+			for _, hit := range expected {
+				if _, ok := resultUIDs[hit.UID]; !ok {
+					d.log.Info("Alternative search query mismatch", "missed", hit.Title, "uid", hit.UID)
+					m++
+				}
+			}
+			if n+m > 0 {
+				d.log.Info("Alternative search query got different results", "matched", (len(expected)+len(results)-n-m)/2, "unexpected", n, "missed", m)
+			}
+		}()
+	}
+
+	start := time.Now()
 	err = d.store.WithDbSession(ctx, func(sess *db.Session) error {
 		return sess.SQL(sql, params...).Find(&res)
 	})
-
+	if d.features.IsEnabled(ctx, featuremgmt.FlagSearchAlt) {
+		d.log.Info("Original search query", "time", time.Since(start), "results", len(res))
+	}
 	if err != nil {
 		return nil, err
 	}
 
+	resc <- res
 	return res, nil
 }
 
+// searchHit represents a single matching row from the search query below.
```

Comment:
Similar comment as above. It would be great if these new structs and everything else that's introduced below could live in a dedicated file which is labeled somehow as belonging to alt search. This is in the hope of making it easier to distinguish behind the different searches.